### DISCUSSION TITLE
Board differences

### DISCUSSION THREAD

**Discussion by The-Futurist** [nanoframework/Home#741] [Q&A] [answered]
Hi,

I can see there is firmware available for the board ST_NUCLEO64_F091RC but is there anything for the ST_NUCLEO64_F446RE?

I'm curious how the firmware would differ between these two boards too, would it be a largely mechanical thing to alter the ST_NUCLEO64_F091RC firmware to create it for the ST_NUCLEO64_F446RE or is it more involved than that?

I'm not too concerned about the board itself personally but I'm curious about the effort required to add support for a board that is similar to an existing supported board.

Also where is the source code etc. for these reference implementations?

Thanks

**Comment by josesimoes** [comment, marked as answer]
The boards for which we have "ready to flash" images are listed [here](https://github.com/nanoframework/Home#firmware-for-reference-boards) and [here](https://github.com/nanoframework/nf-Community-Targets#available-community-boards).

Starting a port to a new board it's relatively easy, specially because there are several existing boards for the STM32F4 series. Just pick one that's closest to your target and you can start from there. 

All the code for the existing boards is at:
https://github.com/nanoframework/nf-interpreter 
and 
https://github.com/nanoframework/nf-Community-Targets
